numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -238,6 +238,9 @@ typedef enum {
 #define NPY_DATETIME_MAX_ISO8601_STRLEN (21+3*5+1+3*6+6+1)
 
 typedef enum {
+        /* Force signed enum type, must be -1 for code compatibility */
+        NPY_FR_ERROR = -1, /* error or undetermined */
```

Comment:
Any idea what FR stands for?

Reply:
Nope, I was wondering the same. This code started as a GSOC project way back when and I assume it meant something at the time. That reminds me,  numpy needs a better datetime and that might be another project going forward.

Reply:
The name was changed [here](f2a723d47b3df1e4f436ccb2524ad3a33a08b8a9)

Reply:
Mystery solved: it's `FRequency`, based on [these lines](https://github.com/numpy/numpy/commit/f2a723d47b3df1e4f436ccb2524ad3a33a08b8a9#diff-cf2eabe7bb5b83d1aa9a0bff1400c1c6R2229)

Reply:
Could you add a comment above to that effect, while you're here?

Reply:
OK, done.